src/deploy/functions/runtimes/node/parseTriggers.ts

```diff
@@ -219,6 +224,41 @@ export function addResourcesToBackend(
       }
       endpoint.vpcConnector = maybeId;
     }
+
+    if (annotation.secrets) {
+      const secretEnvs: backend.SecretEnvVar[] = [];
+      for (const secret of annotation.secrets) {
+        if (secret.includes("/")) {
+          // Assume full secret resource name, either a secret or a secret version name.
+          //   (e.g. projects/my-project/secrets/MY_API_KEY/versions/2)
+          const parsedSecret = parseSecretResourceName(secret);
+          const secretEnv: backend.SecretEnvVar = {
+            key: parsedSecret.name,
+            secret: parsedSecret.name,
+            projectId: parsedSecret.projectId,
+          };
+          if (secret.includes("/versions/")) {
+            const secretVersion = parseSecretVersionResourceName(secret);
+            secretEnv.version = secretVersion.version;
+          }
+          secretEnvs.push(secretEnv);
+        } else {
+          // Assume secret resource id (e.g. MY_API_KEY[@VERSION]).
+          const [key, version] = secret.split("@");
+          const secretEnv: backend.SecretEnvVar = {
+            key,
```

Comment:
It looks like we set `key` & `secret` to the same value in both if cases, why do we need both of them?

Reply:
For context, key = name of the environment variable and secret is the name of the secret.

We happen to enforce that `key == secret` in CF3, but I don't think we necessarily need that to be case for all backend. I'm leaning into "manifest should be general, our SDK should be opinionated" the idea.

E.g. if you are hand-writing stack.yaml, I think it's nice to have the flexibility of renaming a secret name to a different env var key.

WDYT?

Reply:
Yeah I like that take, thanks for clarifying!